applications/FluidDynamicsApplication/python_scripts/y_plus_output_process.py

```diff
@@ -0,0 +1,184 @@
+
+                file_header = self._GetFileHeader()
+                self.output_file = TimeBasedAsciiFileWriterUtility(self.model_part,
+                                                                   file_handler_params, file_header).file
+
+        self.distribution_params = Kratos.Parameters('''{
+            "number_of_value_groups" : 1,
+            "min_value"              : 0.0,
+            "max_value"              : "max"
+        }''')
+        self.distribution_params["max_value"].SetDouble(self.y_plus_output_limit)
+
+
+        compute_y_plus_process_settings = Kratos.Parameters("""{
+            "model_part_name"                  : "",
+            "output_variable_name"             : "Y_PLUS",
+            "output_to_elements"               : false,
+            "calculate_normals_every_time_step": false,
+            "echo_level"                       : 0
+        }""")
+        compute_y_plus_process_settings["model_part_name"].SetString(params["model_part_name"].GetString())
+        compute_y_plus_process_settings["output_to_elements"].SetBool(params["output_to_elements"].GetBool())
+        compute_y_plus_process_settings["calculate_normals_every_time_step"].SetBool(params["calculate_normals_every_time_step"].GetBool())
+        compute_y_plus_process_settings["echo_level"].SetInt(params["echo_level"].GetInt())
+        self.compute_y_plus_process = KratosCFD.ComputeYPlusProcess(model, compute_y_plus_process_settings)
+
+    def Check(self):
+        return self.compute_y_plus_process.Check()
+
+    def ExecuteInitializeSolutionStep(self):
+        if self.IsOutputStep():
+            self.compute_y_plus_process.ExecuteInitializeSolutionStep()
+
+    def ExecuteFinalizeSolutionStep(self):
+        if self.IsOutputStep():
+            self.compute_y_plus_process.ExecuteFinalizeSolutionStep()
+
+    def PrintOutput(self):
+        output = self._CalculateWithRespectToThreshold()
+
+        if (self.model_part.GetCommunicator().MyPID() == 0):
+            current_time = self.model_part.ProcessInfo[Kratos.TIME]
+            output_vals = [format(val, self.format) for val in output]
+
+            # not formatting time in order to not lead to problems with time recognition
+            # in the file writer when restarting
+            output_vals.insert(0, str(current_time))
+
+            res_labels = ["time: ", "mean: ", "std: ", "max: ", "y_plus > " +
+                            "{:.1f}".format(self.y_plus_output_limit) + ": "]
+
+            if (self.print_to_screen):
+                result_msg = 'y+ evaluation for model part ' + \
+                    self.model_part_name + '\n'
+                result_msg += ', '.join([a+b for a,
+                                            b in zip(res_labels, output_vals)])
+                self._PrintToScreen(result_msg + "%")
+
+            if (self.write_output_file):
+                self.output_file.write(' '.join(output_vals) + "\n")
```

Comment:
We're gonna have to change ```TimeBasedAsciiFileWriterUtility``` to support appending files instead of keeping them open for the entire duration of the simulation at some point.

Reply:
> We're gonna have to change `TimeBasedAsciiFileWriterUtility` to support appending files instead of keeping them open for the entire duration of the simulation at some point.

For this actually I think of it as following: ASCII, VTK and GID output is good for fast postprocessing, even though perhaps not so efficient (either in size or having it open - general file handling). For everything else, there should be for example H5 output.